tensorflow_datasets/core/download/downloader_test.py

```diff
@@ -144,6 +143,15 @@ def test_ftp_error(self):
     promise = self.downloader.download(url, self.tmp_dir)
     with self.assertRaises(downloader.urllib.error.URLError):
       promise.get()
+  
+  def test_ssl_mock(self):
+    ssl_mock_dict = downloader.ssl.__dict__.copy()
```

Comment:
This test doesn't seem to be verifying the feature that you've added.

(for example - you should somehow check that the call to fetch the request did contain the "context" parameter etc.

Did you also do manual verification ?

Reply:
I just checked if it is throwing any error for python 2.7.8 flow. I'm not really sure how to mimic SSLError. with either requests or urllib. Can you help me out with that?

Reply:
For starters: the path that you're requesting in this test, needs to be ftp (otherwise your code is not executed).

Then, you have to mock/set side effects on the urllib part (see the test_ftp methods).

And afterwards you can verify that the urlopen was called with the context object, by using
assert_called_once_with

Reply:
That sounds great! Thanks!